lib/importer.py

```diff
@@ -852,6 +889,27 @@ def execImport(handle: int, options: dict):
         log(f"cannot retrieve {mediaTypes} items without any library section", xbmc.LOGERROR)
         return
 
+    # Decide if doing fast sync or not, if so set filter string to include updatedAt
+    fastSync = True
+    lastSync = mediaImport.getLastSynced()
+    updatedFilterString = ""
+    watchedFilterString = ""
+
+    # Check if import settings have changed, or if this is the first time we are importing this library type
+    if not lastSync:
+        fastSync = False
+        log("first time syncronizing library, forcing a full syncronization", xbmc.LOGINFO)
+
+    if SynchronizationSettings.HaveChanged(importSettings=importSettings, providerSettings=providerSettings, save=True):
```

Comment:
Shouldn't this be an `elif` or at least check if `fastSync` is already `False`? There's no reason to check this if `lastSync` is not set.

Reply:
I had made it a separate if intentionally so that on first import of a library the hash still gets generated. I'll put it back to an elif and throw a call to calculate hash under the first run check.